test/cypress/e2e/support/start-app-instance.js

```diff
@@ -12,6 +12,25 @@ Cypress.Commands.add('startAppInstance', (appName, anonymous=false) => {
     },
   };
 
+  if (Cypress.env('responseFuzzing') === 'on') {
+    const [min, max] = [10, 1000];
+    cy.log(`Response fuzzing on, will delay responses randomly between ${min}ms and ${max}ms`);
```

Comment:
Just wondering if this will "lead to more flaky tests". What I mean by that is, since its random, the tests will pass sometimes (when delay is 10ms) and fail other times (when delay is 1000ms). Would it be better/less noisy to just use a slow variant (always 1000ms) all the time?

If we have some conditions where a slow response causes issues, we should be able to reliably reproduce and fix it.

Reply:
Heh, introducing flakiness is the whole point.. :wink: In principle, our app should work the exact same regardless of the backend using more or less time to handle an API request, but that's not always the case.

Let's say you have some code firing off request `A` to do something simple, then it fires off request `B` to do something more complex. Every time you run this locally, `A` will finish before `B`, so you end up unintentionally writing your code in a way where the reducer for `BFulfilled` depends on data from the `AFulfilled` reducer which is already in the store. If we use a random delay, `B` will sometimes finish before `A`, and this fuzzing would then show us an actual bug in our code we'd need to fix. If, however, we just use a fixed delay of 1000ms, `A` will still always finish before `B`, so no problems are found.

So, seemingly random response times from the server is what happens in the real world, and it should not affect our app at all. If/when it does, that shouldn't be considered a flaky test - it's actually a bug in our code, and we need to write more defensive code to make sure everything works as expected even when some requests (but not all) take long to complete.

Reply:
Btw, I don't think we're ready to run these nightly. I ran the `frontend-test` suite with fuzzing once on friday, and multiple tests failed - indicating we still have some of these bugs left. I'd like to clean those up before starting to run fuzzing tests periodically.

Reply:
Right I see what you mean. Makes sense to keep some randomisations then. But perhaps the values should be a bit higher in general(f.ex 500ms for min and 2500 for max?) so that cases where any slow response in general will fail consistenly?

Reply:
Yep, that's a good point! It will take a looong time to run tests on github actions then, though - there are many requests, so this quickly adds up. I'll do a few test runs so I can get some numbers on the effects of this.

Reply:
Not rush to change this until we start running this nightly though, more important to get this merged to unblock other PRs 🙂 We can always adjust this timeout later when we get some experience with it

Reply:
Results are in!

Without fuzzing: **10min 39sec** (all tests successful)
Fuzzing (10-1000ms): **22min 54sec** (`components.js`, `group.js` and `validation.js` each failed once)
Fuzzing (500-2500ms): **40min 54sec** (`components.js`, `group.js`, `validation.js` and `wcag.js` each failed once)

As this is supposed to be random, the timings are, of course, approximate. And again, as it's random, which tests fail when may also be random. Maybe I should just make the min/max configurable..